src/utils.ts

```diff
@@ -331,10 +331,8 @@ export function assertNoMissingTypes(schema: GraphQLSchema, missingTypes: Record
   }
 }
 
-export function assertAbstractTypesCanBeDiscriminated(schema: NexusGraphQLSchema, features: NexusFeatures) {
-  // If backing type check is enabled, we can no longer know for sure if a type is properly discriminated
-  // since it could be discriminated via the `__typename` field in resolvers, which we can't check at runtime
-  if (features.abstractTypeStrategies.__typename === true) {
+export function runAbstractTypeRuntimeChecks(schema: NexusGraphQLSchema, features: NexusFeatures) {
+  if (!features.abstractTypeRuntimeChecks) {
     return
   }
```

Comment:
I think the `__typename` strategy should still be disabling the runtime checks, even if that's confusing. Otherwise, we'll have false reports

Reply:
Yeah thought about it. Should the source of truth be the config or arbitrary logic anywhere in the coebase (don't mean that negatively)?

Reply:
Ohhhh, I did not see this was handled in the `setConfigDefaults` function. Yeah, I frankly don't know the best approach here

Reply:
I'll leave it the way you did!

Reply:
Dumb settings layer and smart components or smart settings layer and dumb components.

I think I'm in favour of smart settings layer because it can be thought of as an integration layer. Each component then doesn't need to know about each other. If there are rules about how they integrate, that can be done higher level rather than violating the logical encapsulation rules.

This is actually one of the ideas driving `setset`.

Once settings have been resolved, the final config is a kind of "absolute source of truth" that drives the rest of the system.

This design is arguably more functional in the sense that it is closer to an assembly line of transformations.